compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/converters/URIDryConverter.java

```diff
@@ -98,73 +100,77 @@ public URIDryConverter(URI base, Map<PackageID, Manifest> dependencyManifests, b
             // ignore errors
         }
     }
-    
+
     public Stream<CompilerInput> finalize(URI remoteURI, PackageID moduleID) {
         try {
             // only continue if a fixed module version is not set. a module version may be set through Ballerina.toml or
             // Ballerina.lock already.
             Matcher matcher = semVerPatchPattern.matcher(moduleID.version.value);
             if ("".equals(moduleID.version.value) || "*".equals(moduleID.version.value) || matcher.matches()) {
-                for (String supportedPlatform : supportedPlatforms) {
-                    HttpURLConnection conn;
-                    // set proxy if exists.
-                    if (null == this.proxy) {
-                        conn = (HttpURLConnection) remoteURI.toURL().openConnection();
-                    } else {
-                        conn = (HttpURLConnection) remoteURI.toURL().openConnection(this.proxy);
-                    }
-                    conn.setInstanceFollowRedirects(false);
-                    conn.setRequestMethod("GET");
-                    // set implementation version
-                    conn.setRequestProperty("Ballerina-Platform", supportedPlatform);
-                    conn.setRequestProperty("Ballerina-Language-Specification-Version", IMPLEMENTATION_VERSION);
-                    
-                    // status code and meaning
-                    //// 302 - module found
-                    //// 404 - module not found
-                    //// 400 - bad request sent
-                    //// 500 - backend is broken
-                    int statusCode = conn.getResponseCode();
-                    if (statusCode == 302) {
-                        // get the version from the 'Location' header.
-                        String location = conn.getHeaderField("Location");
-                        String version = location.split("/")[location.split("/").length - 3];
-                        // update version of moduleID
-                        moduleID.version = new Name(version);
-                        return Stream.empty();
-                    } else if (statusCode == 400 && !loggedError) {
-                        try (BufferedReader errorStream = new BufferedReader(
-                                new InputStreamReader(conn.getInputStream(), Charset.defaultCharset()))) {
-                            String errorContent = errorStream.lines().collect(Collectors.joining("\n"));
-                            this.errStream.println("error: could not connect to remote repository to find the latest " +
-                                                   "version of module: " + moduleID.toString() + ". use '--offline' " +
-                                                   "flag to build in offline mode. reason: " + errorContent);
-                            setErrorLoggedStatusAsTrue();
+                HttpURLConnection conn;
```

Comment:
With this change seems we are not using the `List<String> supportedPlatforms` anymore?